### DISCUSSION TITLE
Support single python file script dependencies PEP722

### DISCUSSION THREAD

**Discussion by nikolaik** [renovatebot/renovate#32279] [Suggest an Idea]
### Tell us more.

Hi, [uv](https://github.com/astral-sh/uv/) has support for [PEP722](https://peps.python.org/pep-0722/) dependencies. Would be really great if renovate would keep those up to date.

Example `yolo.py`:
```py
# /// script
# requires-python = ">=3.9"
# dependencies = [
#     "codeowners==0.7.0",
# ]
# ///
from pathlib import Path
from codeowners import CodeOwners

if __name__ == "__main__":
    owners = CodeOwners(Path(".github/CODEOWNERS").read_text())
    print(owners.of("README.md"))
```
which can be run using

```sh
uv run yolo.py
```

**Comment by rarkins** [comment]
Is this the same as https://github.com/renovatebot/renovate/discussions/27889 ?

**Reply by nikolaik** [reply]
Ah, yes, mine is a duplicate! I'm coming from uv usage of the standard. Is there interested in a PR with support for this?

**Reply by mkniewallner** [reply]
It has already been implemented in https://github.com/renovatebot/renovate/pull/31266. Documentation for it is [here](https://docs.renovatebot.com/modules/manager/pep723/). btw this is [PEP 723](https://peps.python.org/pep-0723/), which has replaced PEP 722 and was the one that was accepted.

**Reply by nikolaik** [reply]
> It has already been implemented in https://github.com/renovatebot/renovate/pull/31266. Documentation for it is [here](https://docs.renovatebot.com/modules/manager/pep723/). btw this is [PEP 723](https://peps.python.org/pep-0723/), which has replaced PEP 722 and was the one that was accepted.

Aha, that's why I did not find things when searching, thanks 👏
